searx/webapp.py

```diff
@@ -896,19 +897,28 @@ def __get_translated_errors(unresponsive_engines: Iterable[UnresponsiveEngine]):
 @app.route('/about', methods=['GET'])
 def about():
     """Redirect to about page"""
-    return redirect(url_for('help_page', pagename='about'))
+    return redirect(url_for('doc', pagename='about'))
+
+
+_ONLINE_DOC = _doc.OnlineDoc(_doc.Mistletoe())
 
 
-@app.route('/help/en/<pagename>', methods=['GET'])
-def help_page(pagename):
-    """Render help page"""
-    page = user_help.PAGES.get(pagename)
+@app.route('/doc/<pagename>', methods=['GET'])
+def doc(pagename):
+    """Render page of online user documentation"""
 
+    locale = request.preferences.get_value('locale')
+    page = _ONLINE_DOC.render(pagename, locale)
+    if page is None:
+        page = _ONLINE_DOC.render(pagename)
     if page is None:
         flask.abort(404)
 
     return render(
-        'help.html', page=user_help.PAGES[pagename], all_pages=user_help.PAGES.items(), page_filename=pagename
+        'doc.html',
+        all_pages=_ONLINE_DOC.all_pages,
```

Comment:
The call to `all_pages` is going to render all the pages for the selected language.

Why not make the code simpler:
* pre-render all the pages when the application starts.
* server the rendered page from this endpoint

Reply:
> The call to `all_pages` is going to render all the pages for the selected language.

Yes, we need the titles (in the tabs) of all pages in the selected languages.

> pre-render all the pages when the application starts.

IMO rendering Markdown to HTML is an *expensive* task .. the parser needs to be started and the output (of all pages in all languages) need to be rendered.  I think a lazy rendering on demand combined with caching is much *cheaper* since users will not visit very often the online documentation and IMO no user will read all online doc pages in all languages.

Reply:
> IMO rendering Markdown to HTML is an expensive task ..

To have an idea, https://github.com/jlevy/the-art-of-command-line/blob/master/README.md (40KB) takes few seconds on my laptop (slow mode, power cable unplugged). Multiple by the number of the language this can be a problem if we have the same amount of document (once again, we don't know what will be the user documentation which makes the decision taking not easy).

The rendering time issue exists also in this PR, because any request to the documentation is going to render the all pages  for this language.

Call sequence:
* https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/webapp.py#L907
* https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/doc.py#L266
* https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/webapp.py#L917-L922
* https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/templates/simple/doc.html#L5
* https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/doc.py#L260
* for each pages of a language: 
  * https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/doc.py#L263
  * https://github.com/searxng/searxng/blob/5d45cd6eb18095e0bb3e5715c58072e68c4aa727/searx/doc.py#L266 

Idea:
* load the raw content of all the pages
* parse the raw content to get the title and the markdown content
* then do lazy rendering of the markdown content

So we have real lazy rendering.

Reply:
> So we have real lazy rendering.

Sound good to me, I will give it a try in a rewrite of this PR.